dimscord/dispatch.nim

```diff
@@ -9,6 +9,17 @@ when (NimMajor, NimMinor, NimPatch) >= (1, 6, 0):
 when defined(dimscordVoice):
     from voice import pause, disconnect
 
+proc checkIfAwaiting(client: DiscordClient, event: DispatchEvent, data: DimscordObject) =
+  ## Runs `data` against a series of handlers waiting on `id`.
+  # TODO: Using pointer so works same as when I add the inner table
+  var handlers = addr client.waits[event]
+  # We countdown so we can delete while iterating
+  for i in countdown(handlers[].len - 1, 0):
+    if handlers[i](data):
+      # Remove the handler if it gets completed
+      handlers[].del(i)
+  echo client.waits[event].len
```

Comment:
Not sure if that’s intentional or..?

Reply:
echo is not intentional whoops. Will remove